### ISSUE TITLE
Remove `QueryRequest` from `GreptimeRequest`

### ISSUE THREAD

**Issue by MichaelScofield** [GreptimeTeam/greptimedb#3799]
### What type of enhancement is this?

Refactor

### What does the enhancement do?

We have the `QueryRequest` in `GreptimeRequest`:

```protobuf
message GreptimeRequest {
  RequestHeader header = 1;
  oneof request {
    InsertRequests inserts = 2;
    QueryRequest query = 3; <-- here
    DdlRequest ddl = 4;
    DeleteRequests deletes = 5;
    RowInsertRequests row_inserts = 6;
    RowDeleteRequests row_deletes = 7;
  }
}
```

However, now the `GreptimeRequest` is not supposed to be used for query. So the `QueryRequest` in it (but not the `QueryRequest` itself!) can be deleted, along with some codes.

**Comment by MichaelScofield** [comment]
@tisonkun right, but normally we first make the changes to proto at our own fork, then change the db codes, and finally merge the proto.

**Comment by tisonkun** [comment]
I found same of the dependencies to QueryRequest is from our inner client `Database`, that is related to #3798.

So this issue can be blocked from #3798.
